src/SprykerEco/Zed/AvalaraTaxShipmentTax/Business/Calculator/MultiShipmentAbstractShipmentAvalaraTaxCalculator.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\AvalaraTaxShipmentTax\Business\Calculator;
+
+use ArrayObject;
+use Generated\Shared\Transfer\AvalaraCreateTransactionResponseTransfer;
+use Generated\Shared\Transfer\AvalaraTransactionLineTransfer;
+use Generated\Shared\Transfer\AvalaraTransactionTransfer;
+use Generated\Shared\Transfer\CalculableObjectTransfer;
+use Generated\Shared\Transfer\ShipmentGroupTransfer;
+use SprykerEco\Zed\AvalaraTaxShipmentTax\Business\Mapper\AvalaraLineItemMapper;
+
+class MultiShipmentAbstractShipmentAvalaraTaxCalculator extends AbstractShipmentAvalaraTaxCalculator
+{
+    /**
+     * @param \Generated\Shared\Transfer\CalculableObjectTransfer $calculableObjectTransfer
+     * @param \Generated\Shared\Transfer\AvalaraCreateTransactionResponseTransfer $avalaraCreateTransactionResponseTransfer
+     *
+     * @return \Generated\Shared\Transfer\CalculableObjectTransfer
+     */
+    public function calculateTax(
+        CalculableObjectTransfer $calculableObjectTransfer,
+        AvalaraCreateTransactionResponseTransfer $avalaraCreateTransactionResponseTransfer
+    ): CalculableObjectTransfer {
+        $regionZipCodeMap = $this->getRegionZipCodeMap($avalaraCreateTransactionResponseTransfer->getTransactionOrFail());
+        $shipmentGroupTransfers = $this->shipmentService->groupItemsByShipment($calculableObjectTransfer->getItems());
+
+        foreach ($shipmentGroupTransfers as $shipmentGroupTransfer) {
+            if (!$shipmentGroupTransfer->getShipment() || !$shipmentGroupTransfer->getShipmentOrFail()->getMethod()) {
+                continue;
+            }
+
+            $avalaraTransactionLineTransfer = $this->findAvalaraLineItemTransferForShipmentGroupTransfer(
+                $shipmentGroupTransfer,
+                $avalaraCreateTransactionResponseTransfer->getTransactionOrFail()->getLines(),
+                $regionZipCodeMap
+            );
+
+            if (!$avalaraTransactionLineTransfer) {
+                continue;
+            }
+
+            $taxRate = $this->sumTaxRateFromTransactionLineDetails($avalaraTransactionLineTransfer->getDetailsOrFail());
+            $taxAmount = $this->moneyFacade->convertDecimalToInteger($avalaraTransactionLineTransfer->getTaxOrFail()->toFloat());
+
+            if ($calculableObjectTransfer->getOriginalQuote()) {
+                $this->setQuoteExpenseTax(
```

Comment:
* Use expenses from `$calculableObjectTransfer->getExpenses()`, because in other cases your calculation plugin become quote specific and will not work for the order.

Reply:
Resolved.